src/modules/service/VoteraService.ts

```diff
@@ -0,0 +1,178 @@
+
+    This file contains the implementation of data collection service from votera.
+
+    Copyright:
+        Copyright (c) 2020-2021 BOSAGORA Foundation
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+import * as cron from "node-cron";
+import Stoa from "../../Stoa";
+import { VoteraClient } from "../votera/VoteraClient";
+import { URL } from "url";
+import { IMetaData, IPendingProposal } from "../../Types";
+import { logger } from "../common/Logger";
+import { Operation } from "../common/LogOperation";
+import { HeightManager } from "../common/HeightManager";
+
+export class VoteraService {
+
+    /**
+     * The Votera endpoint
+     */
+    public votera: VoteraClient;
+
+    /**
+      * Job to collect Unprocessed Proposals data.
+      */
+    private job: cron.ScheduledTask | null = null;
+
+    /**
+     * Job execution status
+     */
+    private status: boolean;
+
+    constructor(votera_endpoint: URL) {
+        this.votera = new VoteraClient(votera_endpoint)
+        this.status = false;
+    }
+
+    /**
+     * Asynchronously start Proposal Data sync job. Job is scheduled fixed @time interval.
+     * @param StoaInstance
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called and if an error occurs the `.catch`
+     * is called with an error.
+     */
+    public start(stoaInstance: Stoa, time: number = 15): Promise<boolean> {
+        return new Promise<boolean>((resolve, reject) => {
+            this.job = cron.schedule(`*/${time} * * * * *`, async () => {
+                await this.scheduler(stoaInstance);
+                resolve(true);
+            });
+        });
+    }
+
+    /**
+     * This method Handles the Proposal data syncing.
+     * The execution order of cron job. 
+     * @param stoaInstance Instance of stoa.
+     * @returns Returns the Promise. If it is finished successfully the `.then`
+     * of the returned Promise is called and if an error occurs the `.catch`
+     * is called with an error.
+     */
+    public scheduler(stoaInstance: Stoa): Promise<boolean> {
+        return new Promise<boolean>(async (resolve, reject) => {
+            try {
+                if (this.status === true) {
+                    logger.info(`votera proposal data collection job is already running.`);
+                    return resolve(true);
+                }
+
+                else {
+                    logger.info(`***votera proposals data syncing job started***`);
+                    this.status = true;
+                    const proposal = await stoaInstance.getPendingProposal();
+                    await Promise.all(proposal.map(async (element) => {
+                        let metadata = await this.getMetadata(element);
+                        await this.storeMetadata(stoaInstance, metadata);
```

Comment:
The test case for these two function calls is missing.
`getMetadata`, `storeMetadata`

Reply:
@TrustHenry I added testcase for getMetadata. i will add the testcases for storeMetadata while creating get API for Proposal.